src/settings.rs

```diff
@@ -0,0 +1,17 @@
+use SettingsExt;
+
+pub trait SettingsExtManual<T> {
+    fn get(&self, key: &str) -> T;
+
+    fn set(&self, key: &str, value: T) -> bool;
+}
+
+impl<bool, O: SettingsExt> SettingsExtManual<bool> for O {
```

Comment:
You need a custom trait here for the type, something like

```rust
pub trait SettingsValueType {
    fn get<T: IsA<Settings>>(&T) -> Self;
    fn set<T: IsA<Settings>>(&T, &self);
}

impl SettingsValueType for bool {
    [...]
}

impl<O: SettingsExt> SettingsExtManual for O {
    fn get<T: SettingsValueType>(&self, key: &str) -> T {
        <T as SettingsValueType>::get(self)
    }

    fn set<T: SettingsValueType>(&self, key: &str, value: &T) {
        <T as SettingsValueType>::set(self, value)
    }
}
```

Reply:
Oh, alright, thanks

Reply:
Done